src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2092,17 +2104,33 @@ public override void AfterTransactionBegin(ITransaction tx)
 			using (new SessionIdLoggingContext(SessionId))
 			{
 				CheckAndUpdateSessionStatus();
-				Interceptor.AfterTransactionBegin(tx);
+
+				if (!_transactionCoordinatorShared || Interceptor != ConnectionManager.Session.Interceptor)
```

Comment:
Ok, I gave it another though. I think that this code makes `ISharedSessionBuilder.Interceptor()` to have no sense at all, as shared interceptor would never be called.